lib/screens/authentication/recover/recover_account_success/interactor/viewmodels/recover_account_success_bloc.dart

```diff
@@ -20,17 +25,38 @@ class RecoverAccountSuccessBloc extends Bloc<RecoverAccountSuccessEvent, Recover
     on<FetchInitialData>(_fetchInitialData);
     on<OnRefreshTapped>(_onRefreshTapped);
     on<OnRecoverFundsTapped>(_onRecoverFundsTapped);
+    on<OnCleanupAndRemoveTapped>(_onCleanupAndRemoveTapped);
+    on<OnRemoveGuardiansConfigTapped>(_onRemoveGuardiansConfigTapped);
+    on<OnRemoveActiveRecoveryTapped>(_onRemoveActiveRecoveryTapped);
   }
 
   Future<void> _fetchInitialData(FetchInitialData event, Emitter<RecoverAccountSuccessState> emit) async {
-    final Result<BalanceModel> result = await FetchRecoverAccountSuccessDataUseCase().run(state.lostAccount);
     emit(state.copyWith(pageState: PageState.loading));
 
-    if (result.isValue) {
-      final data = result.asValue!.value;
+    final res = await Future.wait([
```

Comment:
all this calls/logic stuff should be in a use case and returning a Data Object back to the Bloc with all the data it needs.

Reply:
Actually I am not really clear how to do this

The use case should return a future <result> right? 

We have a list of Result from all these different calls

So either we return a composite data object that contains the results - but it's not itself a result. 

Or we return a list of Result [] ?